ivi-layermanagement-api/ilmControl/include/ilm_control.h

```diff
@@ -376,6 +397,27 @@ ilmErrorTypes ilm_takeScreenshot(t_ilm_uint screen, t_ilm_const_string filename)
  */
 ilmErrorTypes ilm_takeSurfaceScreenshot(t_ilm_const_string filename, t_ilm_surface surfaceid);
 
+/**
+ * \brief Take a screenshot of a certain surface with callbacks
+ * The screenshot is saved as bmp file with the corresponding filename in case
+ * filename is non-NULL. In case callback_done or callback_error are non-NULL,
+ * they are called when the screenshot has been acquired and in case failure
+ * to acquire screenshot occured respectively.
+ * \ingroup ilmControl
+ * \param[in] filename Optional location where the screenshot should be stored,
+ * if the filename is empty and callbacks are available, function will not blocks
+ * \param[in] surfaceid Identifier of the surface to take the screenshot of
+ * \param[in] callback_done Optional callback called when screenshot is acquired
+ * \param[in] callback_error Optional callback called when screenshot acqusition failed
+ * \param[in] user_data Optional callback user data passed in by called
+ * \return ILM_SUCCESS if the method call was successful
+ * \return ILM_FAILED if the client can not call the method on the service.
+ */
+ilmErrorTypes ilm_takeSurfaceScreenshotWithCallbacks(t_ilm_const_string filename, t_ilm_surface surfaceid,
```

Comment:
I think API becomes a bit more complicated with filename in place. screenshot fd is given to application and application can dump it to a file or process it. Additional encoding to a file is already supported by another API. Application can read and encode to a file. I do not see a need to provide a union of multiple functionalities in one API and make it complex.

My proposal would be remove filename from the argument.